collector/indices.go

```diff
@@ -0,0 +1,182 @@
+					return float64(indexStats.Primaries.Docs.Count)
+				},
+				Labels: defaultIndexLabelValues,
+			},
+			{
+				Type: prometheus.GaugeValue,
+				Desc: prometheus.NewDesc(
+					prometheus.BuildFQName(namespace, "indices", "store_size_bytes_primary"),
+					"Current total size of stored index data in bytes which only primary shards on all nodes",
+					defaultIndexLabels, nil,
+				),
+				Value: func(indexStats IndexStatsIndexResponse) float64 {
+					return float64(indexStats.Primaries.Store.SizeInBytes)
+				},
+				Labels: defaultIndexLabelValues,
+			},
+			{
+				Type: prometheus.GaugeValue,
+				Desc: prometheus.NewDesc(
+					prometheus.BuildFQName(namespace, "indices", "store_size_bytes_total"),
+					"Current total size of stored index data in bytes which all shards on all nodes",
+					defaultIndexLabels, nil,
+				),
+				Value: func(indexStats IndexStatsIndexResponse) float64 {
+					return float64(indexStats.Total.Store.SizeInBytes)
+				},
+				Labels: defaultIndexLabelValues,
+			},
+		},
+	}
+}
+
+func (i *Indices) Describe(ch chan<- *prometheus.Desc) {
+	for _, metric := range i.indexMetrics {
+		ch <- metric.Desc
+	}
+	ch <- i.up.Desc()
+	ch <- i.totalScrapes.Desc()
+	ch <- i.jsonParseFailures.Desc()
+}
+
+func (c *Indices) fetchAndDecodeIndexStats() (indexStatsResponse, error) {
+	var isr indexStatsResponse
+
+	if c.exportIndices {
+		u := *c.url
+		u.Path = "/_all/_stats"
+
+		res, err := c.client.Get(u.String())
+		if err != nil {
+			return isr, fmt.Errorf("failed to get index stats from %s://%s:%s/%s: %s",
+				u.Scheme, u.Hostname(), u.Port(), u.Path, err)
+		}
+		defer res.Body.Close()
+		if res.StatusCode != http.StatusOK {
+			return isr, fmt.Errorf("HTTP Request failed with code %d", res.StatusCode)
+		}
+
+		if err := json.NewDecoder(res.Body).Decode(&isr); err != nil {
+			c.jsonParseFailures.Inc()
+			return isr, err
+		}
+	}
+
+	return isr, nil
+}
+
+func (i *Indices) Collect(ch chan<- prometheus.Metric) {
+	i.totalScrapes.Inc()
+	defer func() {
+		ch <- i.up
+		ch <- i.totalScrapes
+		ch <- i.jsonParseFailures
+	}()
+
+	// clusterHealth
+	clusterHealth := NewClusterHealth(i.logger, i.client, i.url)
+	clusterHealthResponse, err := clusterHealth.fetchAndDecodeClusterHealth()
+	if err != nil {
+		i.up.Set(0)
+		level.Warn(i.logger).Log(
+			"msg", "failed to fetch and decode cluster health",
+			"err", err,
+		)
+		return
+	}
+
+	// indices
+	indexStatsResponse, err := i.fetchAndDecodeIndexStats()
+	if err != nil {
+		i.up.Set(0)
+		level.Warn(i.logger).Log(
+			"msg", "failed to fetch and decode index stats",
+			"err", err,
+		)
+		return
+	}
+	i.up.Set(1)
+
+	// Index stats
+	for indexName, indexStats := range indexStatsResponse.Indices {
+		for _, metric := range i.indexMetrics {
+			ch <- prometheus.MustNewConstMetric(
+				metric.Desc,
+				metric.Type,
+				metric.Value(indexStats),
+				metric.Labels(clusterHealthResponse.ClusterName, indexName)...,
```

Comment:
copy'n'paste error

Reply:
please review the labels you want to attach to the indices metrics. It seems there is not cluster name available